### PR TITLE
fixing xgq clang-tidy issues

### PR DESCRIPTION
**Pull request by maxzhen**

### CODE REVIEW

File: src/runtime_src/core/include/xgq_impl.h [hunk 1]
```diff
@@ -41,6 +41,7 @@
 #if defined(__KERNEL__)
 # include <linux/types.h>
 #else
+# include <stdbool.h>
```
Comment (stsoe) [inline src/runtime_src/core/include/xgq_impl.h:44]: warning: including `stdbool.h` has no effect in C++; consider removing it [hicpp-deprecated-headers,modernize-deprecated-headers]

```cpp
# include <stdbool.h>
~~~~~~~~~~^~~~~~~~~~~
```

File: src/runtime_src/core/include/xgq_impl.h [hunk 2]
```diff
@@ -232,15 +231,18 @@ static inline bool xgq_can_consume(struct xgq *xgq)
...
+ * xgq_notify_peer_produced(), which then will publish all entries at once to peer.
  */
 
 static inline int xgq_alloc(struct xgq *xgq, bool server, uint64_t io_hdl, uint64_t ring_addr,
 	    size_t *ring_len, uint32_t slot_size, uint64_t sq_produced, uint64_t cq_produced)
 {
 	size_t rlen = *ring_len;
 	uint32_t numslots = XGQ_MIN_NUM_SLOTS;
-	struct xgq_header hdr = { 0 };
+	struct xgq_header hdr;
```
Comment (stsoe) [inline src/runtime_src/core/include/xgq_impl.h:245]: warning: uninitialized record type: `hdr` [cppcoreguidelines-pro-type-member-init,hicpp-member-init]

```cpp
struct xgq_header hdr;
^
                     {}
```

File: src/runtime_src/core/include/xgq_impl.h [hunk 3]
```diff
@@ -278,7 +283,7 @@ static inline int xgq_alloc(struct xgq *xgq, bool server, uint64_t io_hdl, uint6
 static inline int xgq_attach(struct xgq *xgq, bool server, uint64_t ring_addr,
 			     uint64_t sq_produced, uint64_t cq_produced)
 {
-	struct xgq_header hdr = { 0 };
+	struct xgq_header hdr;
```
Comment (stsoe) [inline src/runtime_src/core/include/xgq_impl.h:286]: warning: uninitialized record type: `hdr` [cppcoreguidelines-pro-type-member-init,hicpp-member-init]

```cpp
struct xgq_header hdr;
^
                     {}
```

**Review by stsoe — COMMENTED** [review]
clang-tidy made some suggestions

File: src/runtime_src/core/include/xgq_impl.h (hunk 1 above)
Comment (stsoe) [reply src/runtime_src/core/include/xgq_impl.h:44]: If this header is supposed to be included from C++ code in addition to C code, then maybe fix the include section for C++ within #ifdef __cplusplus ?

Comment (maxzhen) [reply src/runtime_src/core/include/xgq_impl.h:44]: > 
> 
> If this header is supposed to be included from C++ code in addition to C code, then maybe fix the include section for C++ within #ifdef __cplusplus ?

Will try it.

File: src/runtime_src/core/include/xgq_impl.h (hunk 2 above)
Comment (maxzhen) [reply src/runtime_src/core/include/xgq_impl.h:245]: clang-tidy complains about {0}, so this has to be removed. can we tell clang-tidy to ignore this line?

Comment (stsoe) [reply src/runtime_src/core/include/xgq_impl.h:245]: It is clangtidy that added the comment above and the variable should be initialized for sure. You can do just `struct xgq_header hdr {};`  but interestingly, when I run clangtidy  it doesn't complain about = {0}.

Comment (maxzhen) [reply src/runtime_src/core/include/xgq_impl.h]: added struct xgq_header hdr = {}; see if this is OK now?

**Comment by stsoe** [comment]
Changes works, thanks!

Can we fix the include guards please to not use __ (double underscore)?
`.../src/runtime_src/core/include/xgq_impl.h:39:9: warning: declaration uses identifier '__XGQ_IMPL_H__', which is a reserved identifier [bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp]`

**Comment by maxzhen** [comment]
> 
> 
> Changes works, thanks!
> 
> Can we fix the include guards please to not use __ (double underscore)?
> `.../src/runtime_src/core/include/xgq_impl.h:39:9: warning: declaration uses identifier '__XGQ_IMPL_H__', which is a reserved identifier [bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp]`
> 
> clangtidy also complains about line 143 `size_t i,` not being initialized. While it is used as loop variable immediately below, it is still a good idea to initialize variables where they are defined.

fixed.

File: src/runtime_src/core/include/xgq_impl.h [hunk 4]
```diff
@@ -323,15 +330,25 @@ static inline int xgq_produce(struct xgq *xgq, uint64_t *slot_addr)
...
 	if (!xgq_can_consume(xgq))
 		return -ENOENT;
 	ring->xr_consumed++;
 	*slot_addr = xgq_ring_slot_ptr(ring, false);
 
-	/* Make sure this is a new entry */
-	while (!(xgq_mem_read32(xgq->io_hdl, *slot_addr) & XGQ_ENTRY_NEW_FLAG_MASK));
+#ifdef XGQ_OUT_OF_ORDER_WRITE
+	/*
+	 * Producer pointer does not gurantee the slot content is updat-to-date.
```
Comment (larry9523) [inline src/runtime_src/core/include/xgq_impl.h:345]: up-to-date?

**Review by larry9523 — APPROVED** [review]

**Review by stsoe — APPROVED** [review]

**Pull request merged**